core/src/main/java/com/linecorp/armeria/common/Flags.java

```diff
@@ -422,7 +402,32 @@ public static boolean useEpoll() {
      * {@code -Dcom.linecorp.armeria.useOpenSsl=false} JVM option to disable it.
      */
     public static boolean useOpenSsl() {
-        return USE_OPENSSL;
+        if (useOpenSsl != null) {
+            return useOpenSsl;
+        }
+        useOpenSsl = getBoolean("useOpenSsl", OpenSsl.isAvailable(), value -> OpenSsl.isAvailable() || !value);
```

Comment:
I think we can avoid `isAvailable` if the user explicitly disabled OpenSSL.

```java
final boolean useOpenSsl = getBoolean("useOpenSsl", true);
if (!useOpenSsl) {
  // OpenSSL explicitly disabled
  return Flags.useOpenSsl = false;
}
if (!OpenSsl.isAvailable(){ 
  ...
} else {
  ...
}
```

Reply:
Is this useOpenSsl variable unconditionally true?

Reply:
In this case, `true` is the default value unless a user sets `com.linecorp.armeria.useOpenSsl=false`.

At least I think it is let me know if I missed something :)

Reply:
Yes. I understand 😃